src/QMCWaveFunctions/SPOSet.h

```diff
@@ -206,6 +214,18 @@ class SPOSet : public QMCTraits
                                  const ValueVector& psiinv,
                                  std::vector<ValueType>& ratios);
 
+
+  /// Determinant ratios and parameter derivatives of the wavefunction for virtual moves
+  virtual void evaluateDerivRatios(const VirtualParticleSet& VP,
```

Comment:
For both added APIs, I feel it will be better to pass the inverse matrix in instead of recomputing inside. Consider that in a later PR.

Reply:
Agree, as per my comment above.